v1/options.go

```diff
@@ -2,6 +2,166 @@
+//     In contrast, in v2 a Go map is marshaled in a non-deterministic order.
+//     The [jsonv2.Deterministic] option controls this behavior difference.
+//
+//   - In v1, JSON strings are encoded with HTML-specific or JavaScript-specific
+//     characters being escaped. In contrast, in v2 JSON strings use the minimal
+//     encoding and only escape if required by the JSON grammar.
+//     The [jsontext.EscapeForHTML] and [jsontext.EscapeForJS] options
+//     control this behavior difference.
+//
+//   - In v1, bytes of invalid UTF-8 within a string are silently replaced with
+//     the Unicode replacement character. In contrast, in v2 the presence of
+//     invalid UTF-8 results in an error. The [jsontext.AllowInvalidUTF8] option
+//     controls this behavior difference.
+//
+//   - In v1, a JSON object with duplicate names is permitted.
+//     In contrast, in v2 a JSON object with duplicate names results in an error.
+//     The [jsontext.AllowDuplicateNames] option controls this behavior difference.
+//
+//   - In v1, when unmarshaling a JSON null into a non-empty Go value it would
+//     inconsistently either zero out the the value or do nothing.
+//     In contrast, in v2 unmarshaling a JSON null would consistently and always
+//     zero out the underlying Go value. The [MergeWithLegacySemantics] option
+//     controls this behavior difference.
+//
+//   - In v1, when unmarshaling a JSON value into a non-zero Go value,
+//     it merges into the original Go value for array elements, slice elements,
+//     struct fields (but not map values),
+//     pointer values, and interface values (only if a non-nil pointer).
+//     In contrast, in v2 unmarshal merges into the Go value
+//     for struct fields, map values, pointer values, and interface values.
+//     In general, the v2 semantic merges when unmarshaling a JSON object,
+//     otherwise it replaces the value. The [MergeWithLegacySemantics] option
+//     controls this behavior difference.
+//
+//   - In v1, a [time.Duration] is represented as a JSON number containing
+//     the decimal number of nanoseconds. In contrast, in v2 a [time.Duration]
+//     is represented as a JSON string containing the formatted duration
+//     (e.g., "1h2m3.456s") according to [time.Duration.String].
+//     The [FormatTimeWithLegacySemantics] option controls this behavior difference.
+//
+//   - In v1, errors are never reported at runtime for Go struct types
+//     that have some form of structural error (e.g., a malformed tag option).
+//     In contrast, v2 reports a runtime error for Go types that are invalid
+//     as they relate to JSON serialization. For example, a Go struct
+//     with only unexported fields cannot be serialized.
+//     The [ReportErrorsWithLegacySemantics] option controls this behavior difference.
+//
+// As mentioned, the entirety of v1 is implemented in terms of v2,
+// where options are implicitly specified to opt into historical behavior.
```

Comment:
What do you think of "legacy" here? Since all the options use that word.
```suggestion
// where options are implicitly specified to opt into legacy behavior.
```